api/serializers/declaration.py

```diff
@@ -0,0 +1,338 @@
+from rest_framework import serializers
+from api.exceptions import ProjectAPIException
+from drf_base64.fields import Base64FileField
+from data.models import (
+    Declaration,
+    DeclaredPlant,
+    DeclaredMicroorganism,
+    DeclaredIngredient,
+    DeclaredSubstance,
+    ComputedSubstance,
+    Attachment,
+    Condition,
+    Population,
+    Company,
+    PlantPart,
+    Plant,
+    Ingredient,
+    Microorganism,
+    Substance,
+    SubstanceUnit,
+)
+from .plant import PlantSerializer
+from .microorganism import MicroorganismSerializer
+from .ingredient import IngredientSerializer
+from .substance import SubstanceSerializer
+
+
+class IdPassthrough:
+    """
+    Permet de passer l'ID des nested-objects dans les serializers
+    """
+
+    def __init__(self, instance=None, data=..., **kwargs):
+        super().__init__(instance, data, **kwargs)
+        self.fields["id"].read_only = False
+
+
+class PassthroughPlantSerializer(IdPassthrough, PlantSerializer):
+    pass
+
+
+class PassthroughMicroorganismSerializer(IdPassthrough, MicroorganismSerializer):
+    pass
+
+
+class PassthroughIngredientSerializer(IdPassthrough, IngredientSerializer):
+    pass
+
+
+class PassthroughSubstanceSerializer(IdPassthrough, SubstanceSerializer):
+    pass
+
+
+class DeclaredListSerializer(serializers.ListSerializer):
+    """
+    Pour les modèles liés et les list serializers on a besoin de spécifier le comportement
+    dans une mise à jour car DRF ne peut pas le déviner:
+    https://www.django-rest-framework.org/api-guide/serializers/#customizing-multiple-update
+    """
+
+    def update(self, instance, validated_data):
+        instance_mapping = {instance.id: instance for instance in instance}
```

Comment:
`for instance in instance` : pourquoi le nom de la variable de parcours à le même nom que l'iterable parcouru ?